### DISCUSSION TITLE
Give tenant read only access

### DISCUSSION THREAD

**Discussion by mitch2k** [netbox-community/netbox#9445] [General]
Hi,

As an MSP we have multiple tenants in our Netbox install. We want to give customers access to their own tenant so they can see their own data. This is possible with constraints. However, the keyword tenant_id is only valid for device, vlan, etc, not for example cable. This means I have to create 6 permissions for each tenant:

<html>
<body>
<div class="table-wrap" style="margin: 10px 0px 0px; padding: 0px; overflow-x: auto; color: rgb(9, 30, 66); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, &quot;Noto Sans&quot;, Ubuntu, &quot;Droid Sans&quot;, &quot;Helvetica Neue&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">

Name | Models | Groups | Actions | Constraints
-- | -- | -- | -- | --
XXX_readonly_main | dcim \| device, ipam \| VLAN, dcim \| rack, virtualization \| virtual machine, ipam \| IP address, ipam \| prefix, dcim \| site | XXX_readonly | view | [{"tenant_id": 00}]
XXX_readonly_tenant | tenancy \| tenant | XXX_readonly | view | [{"id": 00}]
XXX_readonly_devicedetails | dcim \| front port, dcim \| power outlet, dcim \| interface, dcim \| power port, dcim \| device bay, dcim \| rear port | XXX_readonly | view | [{"device__tenant_id": 00}]
XXX_readonly_cable1 | dcim \| cable | XXX_readonly | view | [{"_termination_a_device__tenant_id": 00}]
XXX_readonly_cable2 | dcim \| cable | XXX_readonly | view | [{"_termination_b_device__tenant_id": 00}]
XXX_readonly_vminterface | virtualization \| interface | XXX_readonly | view | [{"virtual_machine__tenant_id": 00}]

</div><p style="margin: 12px 0px 0px; padding: 0px; color: rgb(9, 30, 66); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, &quot;Noto Sans&quot;, Ubuntu, &quot;Droid Sans&quot;, &quot;Helvetica Neue&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Replace XXX by the tenant name. Replace 00 by the tenant ID.</p>
</body>
</html>

I was wondering if this is the correct way. Is there a way to put this in a single constraint?

Thanks

**Comment by DanSheps** [comment]
> I was wondering if this is the correct way. Is there a way to put this in a single constraint?

Nope, you got the best use of constraints right here I think.

**Comment by uRhaineWork** [comment]
If thats less of a problem You could #tag elements for customers, as tags I think are more uniwersal for components?

**Comment by bscheewe** [comment]
I'm trying to figure out if we can use Netbox in our MSP.  From what I've seen on the demo site, Duplicates, parents, etc show up across regions, sites, or tenants.  How can Netbox be configured to be truly multi-tenanted for an MSP that has many clients.

Sorry if I'm hi-jacking this question, but it's the only place I've seen MSP needs mentioned.

**Reply by a084ed22** [reply]
we have a similar situation of overlapping address ranges across different tenants/entities. The solution we chose is to map each tenant to a separate VRF and keep globally-unique (i.e. public address space) ranges in the global VRF.
